ttnn/cpp/ttnn/tensor/types.cpp

```diff
@@ -293,3 +293,28 @@ MemoryConfig load_memory_config(const std::string& file_name) {
 }  // namespace tt_metal
 
 }  // namespace tt
+
+namespace ttnn {
+namespace types {
+
+uint32_t &Shape::operator[](const std::int64_t index) {
+    auto normalized_index = this->get_normalized_index(index);
+    return this->value[normalized_index];
+}
+
+const uint32_t Shape::operator[](std::int64_t index) const { return this->value.without_padding()[index]; }
```

Comment:
one returns a reference to padded value, another returns a const value of original unpadded value. i don't think its ok

Reply:
this was a tricky one. `this->value.without_padding` is const so I can't use it in the first one. Is there another way to make this work? If we don't have this, I think lots of places will work when they try to modify shape value. I am more in favour of just removing non-const operator, since modifying this shape without considering padding is not good... Most use cases are things like:
- Original unpadded shape
- Modify some dim to be padded
- Return padded shape

But this can either be expressed as:
- Simple shape without padding that we allow to be modified
- Current `ttnn::Shape` (or old `LegacyShape` also works) can express both padding and shape

Reply:
Removing for now. It's giving me bugs when accessing shape that is non-const.